### PR TITLE
Add the basic infrastructure for opening editors for documents

### LINKED ISSUE #37: Implement "open Salt document graph with editor"
- [X] Who is creating the view? **A [commands/handler](https://www.vogella.com/tutorials/EclipseRCP/article.html#commands_parameter) could offer this service for each editor. We could either parametrize the command with the document ID, or use the [global selection API](https://www.vogella.com/tutorials/Eclipse4Services/article.html#selection-service) to get the currently selected document ondemand. The commands can be used by the corpus structure editor [context menu](https://www.vogella.com/tutorials/EclipseRCP/article.html#tutorial_commandscontextmenu).**
- [x] How is it created? **Parts can be created dynamically from templates (`PartDescriptor)`<https://www.vogella.com/tutorials/Eclipse4Services/article.html#example-part-descriptors-and-creating-parts-dynamically>**
- [X] Can a document be open in more than one editor? **Yes, because we have global state (https://github.com/hexatomic/hexatomic/issues/36)**
- [X] Should a document be open in more than one editor? **Possible due to global state (https://github.com/hexatomic/hexatomic/issues/36)**
- [X] How can an editor listen to changes of the graph? **The injectable `ProjectManager` offers a `addListener` function to register a listener.**
- [x] Implement loading
- [x] Implement unloading

### LINKED ISSUE #40: Implement an exemplary editor/view
E.g., a primary text viewer for the selected/opened/whatevered document (depending on #37).

Don't go overboard here, this should be a simple proof-of-concept to test the interfaces, etc.

- [x] Implement in bundle
- [x] Add to core feature
- [x] Document for users

Cf. #39, #37.

### PR DESCRIPTION
**Pull request by thomaskrause**
The corpus structure editor is extended to have a new context menu item "Show Text" which opens a new Part with the text of the selected document. It is based on the PR #22 and therefore is only read for review after #22 has been merged.

A new parametrized command is added to the Hexatomic Core, which allows to open the currently selected document with a new editor given by PartDescriptor ID in the parameter. 
Global selection of documents is implemented in the corpus structure editor and the `OpenSaltDocumentHandler` will only be available if the selection is an actual document.
For illustration, a sample bundle with a simple viewer for the textual datasource has been added.

Currently, the "Show Context" menu item is hard-coded in the application fragment model of the corpus structure editor.
Modifying the application model dynamically to add menu items for each available editors will be part of a future PR that would implement #35.

This PR will closes #37 and closes #40 .

### CODE REVIEW

File: bundles/org.corpus_tools.hexatomic.textviewer/build.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+output.. = target/classes/
+bin.includes = META-INF/,\
+               .,\
+               fragment.e4xmi,\
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply bundles/org.corpus_tools.hexatomic.textviewer/build.properties]: I did not add this manually, but it was created when creating the bundle.

File: docs/user/src/SUMMARY.md [hunk 2]
```diff
@@ -4,4 +4,5 @@
 - [Installation](./installation.md)
 - [Usage](usage/README.md)
 	- [Editing the corpus structure](usage/corpus-structure.md)
+	- [Opening an editor](usage/open-editor.md)
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply docs/user/src/SUMMARY.md]: I disagree. In PR #53 the text of the context menu will already be more general ("Open with XXX editor"). Also, we document what we specified and how our architecture looks like.

File: docs/user/src/usage/open-editor.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Opening an editor
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply docs/user/src/usage/open-editor.md]: See above

File: docs/user/src/usage/open-editor.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+To open an editor for a document of the corpus, first select the corpus in the "Corpus Structure" editor on the left.
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply docs/user/src/usage/open-editor.md]: See above

File: docs/user/src/usage/open-editor.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+After selecting, use the right mouse button to open a context menu and select "Show Text".
+This will open a new view where the text of the document is displayed.
+
+![A view for the text of a document](show-document-text.png)
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply docs/user/src/usage/open-editor.md]: Changed in e4e72ce6e464ccd2add38367d14dae4db8847271

File: bundles/org.corpus_tools.hexatomic.core/src/main/java/org/corpus_tools/hexatomic/core/handlers/OpenSaltDocumentHandler.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+			SDocument document = (SDocument) selection;
+
+			if (document.getDocumentGraph() == null) {
+				if (document.getDocumentGraphLocation() == null) {
+					// create a new document graph, because no one exists yet
+					document.createDocumentGraph();
+				} else {
+					// TODO: show progress indicator
+					document.loadDocumentGraph();
```
[the remark this answers is not in the document]
Comment (thomaskrause) [reply bundles/org.corpus_tools.hexatomic.core/src/main/java/org/corpus_tools/hexatomic/core/handlers/OpenSaltDocumentHandler.java]: Resolved in 22fddf7f03cb07cb0477b7a086c053e72e23e1e4 and 694fdb4bc0f753de901b58391998f9edc34aba59

File: docs/user/src/SUMMARY.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdruskat) [reply docs/user/src/SUMMARY.md]: Right, I see. The traps of stacked PRs...

File: bundles/org.corpus_tools.hexatomic.core/src/main/java/org/corpus_tools/hexatomic/core/handlers/OpenSaltDocumentHandler.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+						// create a new document graph, because no one exists yet
+						document.createDocumentGraph();
+					} else {
+						// TODO: show progress indicator
+						document.loadDocumentGraph();
+					}
+
+				} catch (SaltResourceException ex) {
+					// TODO: display error to the user in a dialog
+					log.error("Could not load document graph (the actual annotations for document {}", document.getId(),
```
Comment (sdruskat) [inline bundles/org.corpus_tools.hexatomic.core/src/main/java/org/corpus_tools/hexatomic/core/handlers/OpenSaltDocumentHandler.java]:
```suggestion
					log.error("Could not load document graph (the actual annotations for document {}).", document.getId(),
```

(Take it you're not yet using ErrorService here as it's not been merged yet.)

**Pull request merged**